lib/spack/spack/fetch_strategy.py

```diff
@@ -502,7 +503,7 @@ def archive(self, destination, **kwargs):
         tar = which('tar', required=True)
 
         patterns = kwargs.get('exclude', None)
```

Comment:
Looking at the code in this function it seems that we can exclude patterns that are unrelated to VCS. In that sense what is written in the docs might be inexact, as one could get those patterns into the archive by giving `clean=False`. Can we target VCS patterns in a more specific way?

Reply:
While it could be used in a more generic way, `exclude` is specific to `VCSFetchStrategy.archive`. Looking around, it seems to be used only by subclasses, and a `git blame` suggests that intent, too. Thus I'd prefer to keep the code simpler…